src/lib.rs

```diff
@@ -175,21 +182,86 @@ enum OfferState {
 }
 
 pub struct OfferHandler {
-    _active_offers: Mutex<HashMap<String, OfferState>>,
+    active_offers: Mutex<HashMap<String, OfferState>>,
     pending_messages: Mutex<Vec<PendingOnionMessage<OffersMessage>>>,
     messenger_utils: MessengerUtilities,
     // This channel will close once the onion messenger has successfully started up.
-    _started: RefCell<Receiver<u32>>,
+    started: RefCell<Receiver<u32>>,
 }
 
 impl OfferHandler {
     pub fn new(started: Receiver<u32>) -> Self {
         OfferHandler {
-            _active_offers: Mutex::new(HashMap::new()),
+            active_offers: Mutex::new(HashMap::new()),
             pending_messages: Mutex::new(Vec::new()),
             messenger_utils: MessengerUtilities::new(),
-            _started: RefCell::new(started),
+            started: RefCell::new(started),
+        }
+    }
+
+    /// Adds an offer to be paid with the amount specified. May only be called once for a single offer.
+    #[allow(clippy::await_holding_refcell_ref)]
+    pub async fn pay_offer(
+        &self,
+        offer: Offer,
+        amount: Option<u64>,
+        network: Network,
+        mut client: Client,
+        blinded_path: BlindedPath,
+        mut reply_path: Option<BlindedPath>,
+    ) -> Result<(), OfferError<Secp256k1Error>> {
+        // Wait for onion messenger to give us the signal that it's ready. Once the onion messenger drops
+        // the channel sender, recv will return None and we'll stop blocking here.
+        while (self.started.borrow_mut().recv().await).is_some() {
+            println!("Error: we shouldn't receive any messages on this channel");
```

Comment:
RefCell seems like it's always more trouble than it's worth - can we just pass the `started` channel in here?